crypto/account.go

```diff
@@ -43,3 +44,72 @@ func GenerateAccount() (kp Account) {
 	kp.Address = a
 	return
 }
+
+/* Multisig Support */
+
+// MultisigAccount is a convenience type for holding multisig preimage data
+type MultisigAccount struct {
+	// Version is the version of this multisig
+	Version uint8
+	// Threshold is how many signatures are needed to fully sign as this address
+	Threshold uint8
+	// Pks is an ordered list of public keys that could potentially sign a message
+	Pks []ed25519.PublicKey
+}
+
+// MultisigAccountWithParams creates a MultisigAccount with the given parameters
+func MultisigAccountWithParams(version uint8, threshold uint8, addrs []types.Address) (ma MultisigAccount, err error) {
+	ma.Version = version
+	ma.Threshold = threshold
+	ma.Pks = make([]ed25519.PublicKey, len(addrs))
+	for i := 0; i < len(addrs); i++ {
+		ma.Pks[i] = addrs[i][:]
+	}
+	err = ma.Validate()
+	return
+}
+
+// MultisigAccountFromSig is a convenience method that creates an account
+// from a sig in a signed tx. Useful for getting addresses from signed msig txs, etc.
+func MultisigAccountFromSig(sig types.MultisigSig) (ma MultisigAccount, err error) {
+	ma.Version = sig.Version
+	ma.Threshold = sig.Threshold
+	ma.Pks = make([]ed25519.PublicKey, len(sig.Subsigs))
+	for i := 0; i < len(sig.Subsigs); i++ {
+		c := make([]byte, len(sig.Subsigs[i].Key))
```

Comment:
is there any particular reason we're creating a copy of these keys, rather than using the input buffer ?

Reply:
It's a little safer, protects against the caller messing around with the keys.